catroid/src/org/catrobat/catroid/devices/arduino/phiro/PhiroImpl.java

```diff
@@ -0,0 +1,384 @@
+		@Override
+		public void run() {
+			sendAnalogFirmataMessage(PIN_SPEAKER_OUT, 0);
+		}
+	}
+
+	@Override
+	public void moveLeftMotorForward(int speedInPercent) {
+		sendAnalogFirmataMessage(PIN_LEFT_MOTOR_FORWARD, percentToSpeed(speedInPercent));
+	}
+
+	@Override
+	public void moveLeftMotorBackward(int speedInPercent) {
+		sendAnalogFirmataMessage(PIN_LEFT_MOTOR_BACKWARD, percentToSpeed(speedInPercent));
+	}
+
+	@Override
+	public void moveRightMotorForward(int speedInPercent) {
+		sendAnalogFirmataMessage(PIN_RIGHT_MOTOR_FORWARD, percentToSpeed(speedInPercent));
+	}
+
+	@Override
+	public void moveRightMotorBackward(int speedInPercent) {
+		sendAnalogFirmataMessage(PIN_RIGHT_MOTOR_BACKWARD, percentToSpeed(speedInPercent));
+	}
+
+	@Override
+	public void stopLeftMotor() {
+		moveLeftMotorForward(0);
+		moveLeftMotorBackward(0);
+	}
+
+	@Override
+	public void stopRightMotor() {
+		moveRightMotorForward(0);
+		moveRightMotorBackward(0);
+	}
+
+	@Override
+	public void stopAllMovements() {
+		stopLeftMotor();
+		stopRightMotor();
+	}
+
+	@Override
+	public void setLeftRGBLightColor(int red, int green, int blue) {
+		red = checkRBGValue(red);
+		green = checkRBGValue(green);
+		blue = checkRBGValue(blue);
+
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_RED_LEFT, red));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_GREEN_LEFT, green));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_BLUE_LEFT, blue));
+	}
+
+	@Override
+	public void setRightRGBLightColor(int red, int green, int blue) {
+		red = checkRBGValue(red);
+		green = checkRBGValue(green);
+		blue = checkRBGValue(blue);
+
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_RED_RIGHT, red));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_GREEN_RIGHT, green));
+		sendFirmataMessage(new AnalogMessage(PIN_RGB_BLUE_RIGHT, blue));
+	}
+
+	private int percentToSpeed(int percent) {
+		if (percent <= 0) {
+			return 0;
+		}
+		if (percent >= 100) {
+			return 255;
+		}
+
+		return (int)(percent * 2.55);
+	}
+
+	private int checkRBGValue(int rgbValue) {
+		if (rgbValue > 255) {
+			return 255;
+		}
+
+		if (rgbValue < 0) {
+			return 0;
+		}
+
+		return rgbValue;
+	}
+
+	@Override
+	public String getName() {
+		return "Phiro";
+	}
+
+	@Override
+	public Class<? extends BluetoothDevice> getDeviceType() {
+		return PHIRO;
+	}
+
+	@Override
+	public void setConnection(BluetoothConnection connection) {
+		this.btConnection = connection;
+	}
+
+	@Override
+	public void disconnect() {
+
+		if (firmata == null) {
+			return;
+		}
+
+		try {
+			resetPins();
+			this.reportSensorData(false);
+			firmata.clearListeners();
+			firmata.getSerial().stop();
+			isInitialized = false;
+			firmata = null;
+		} catch (SerialException e) {
+			Log.d(TAG, "Error stop phiro pro serial");
+		}
+	}
+
+	@Override
+	public boolean isAlive() {
+		if (firmata == null) {
+			return false;
+		}
+
+		try {
+			firmata.send(new ReportFirmwareVersionMessage());
+			return true;
+		} catch (SerialException e) {
```

Comment:
Work in progress: [CAT-1277#comment1](https://jira.catrob.at/browse/CAT-1277?focusedCommentId=16348&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16348)
(Since you may find the naming convention a little tight, also consider to read [CAT-1277#comment2](https://jira.catrob.at/browse/CAT-1277?focusedCommentId=16354&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16354))